### ISSUE TITLE
upgrade from 1.16.0 to 1.17.0, network local is ok, but online fail

### ISSUE THREAD

**Issue by hb-chen** [micro/micro#456]
## v1.16.0
Online, ip:xx.xx.xx.xx
```bash
./micro_1.16.0 network --address=:8085
```
Local
```bash
./micro_1.16.0 network --nodes=xx.xx.xx.xx:8085
./micro_1.16.0 network routes
+------------------+----------------------+----------------------+--------------------------------------+----------+--------+---------+
|     SERVICE      |       ADDRESS        |       GATEWAY        |                ROUTER                | NETWORK  | METRIC |  LINK   |
+------------------+----------------------+----------------------+--------------------------------------+----------+--------+---------+
| go.micro         | 4706869997262573539  | 13031373710906630179 | c299a206-abcc-440a-b952-e97565e5fcd1 | go.micro | 64     | network |
| go.micro         | 192.168.3.232:8085   |                      | 0bc0cfe5-5dfc-4541-8399-87a577c6875c | go.micro | 1      | local   |
| go.micro.network | 14969748229395243695 | 13031373710906630179 | c299a206-abcc-440a-b952-e97565e5fcd1 | go.micro | 64     | network |
| go.micro.network | 192.168.3.232:50862  |                      | 0bc0cfe5-5dfc-4541-8399-87a577c6875c | go.micro | 1      | local   |
+------------------+----------------------+----------------------+--------------------------------------+----------+--------+---------+
```

## v1.17.0
**When I upgrade to 1.17.0 or 1.17.1 faile**

Online
```
./micro_1.17.0 network --address=:8085
```
Local
```
./micro_1.17.0 network --nodes=xx.xx.xx.xx:8085
./micro_1.17.0 network routes
+------------------+---------------------+---------+--------------------------------------+----------+--------+-------+
|     SERVICE      |       ADDRESS       | GATEWAY |                ROUTER                | NETWORK  | METRIC | LINK  |
+------------------+---------------------+---------+--------------------------------------+----------+--------+-------+
| go.micro         | 192.168.3.232:8085  |         | 314a4b33-2c6c-4342-9575-1b4c09ba8933 | go.micro | 1      | local |
| go.micro.network | 192.168.3.232:53731 |         | 314a4b33-2c6c-4342-9575-1b4c09ba8933 | go.micro | 1      | local |
| go.micro.network | 192.168.3.232:53732 |         | 314a4b33-2c6c-4342-9575-1b4c09ba8933 | go.micro | 1      | local |
+------------------+---------------------+---------+--------------------------------------+----------+--------+-------+
```

**But if I run two network on localhost, it's ok.**

Local one
```
./micro_1.17.0 network --address=:8085
```
Local two
```
./micro_1.17.0 --registry=etcd network --nodes=localhost:8085 --address=:9085
/micro_1.17.0 --registry=etcd network routes                                
+------------------+----------------------+---------------------+--------------------------------------+----------+--------+---------+
|     SERVICE      |       ADDRESS        |       GATEWAY       |                ROUTER                | NETWORK  | METRIC |  LINK   |
+------------------+----------------------+---------------------+--------------------------------------+----------+--------+---------+
| go.micro         | 192.168.3.232:9085   |                     | 04d84f2d-395e-4afd-8e31-eb167bd74a7b | go.micro | 1      | local   |
| go.micro         | 16892779280728369246 | 9414556940850352696 | b96dd458-7f74-483f-93d1-baab791631fe | go.micro | 2      | network |
| go.micro.network | 1180513087039894170  | 9414556940850352696 | b96dd458-7f74-483f-93d1-baab791631fe | go.micro | 2      | network |
| go.micro.network | 192.168.3.232:56076  |                     | 04d84f2d-395e-4afd-8e31-eb167bd74a7b | go.micro | 1      | local   |
| go.micro.network | 192.168.3.232:56077  |                     | 04d84f2d-395e-4afd-8e31-eb167bd74a7b | go.micro | 1      | local   |
+------------------+----------------------+---------------------+--------------------------------------+----------+--------+---------+
```

**Comment by hb-chen** [comment]
Replace IP with domain is ok.

**Comment by asim** [comment]
Fixed in 1.18.0
